app/controllers/admin/secrets_controller.rb

```diff
@@ -13,9 +13,13 @@ class Admin::SecretsController < ApplicationController
   before_action :authorize_any_deployer!, only: DEPLOYER_ACCESS
 
   def index
-    @secret_keys = SecretStorage.keys
-    if query = params.dig(:search, :query).presence
-      @secret_keys.select! { |s| s.include?(query) }
+    begin
+      @secret_keys = SecretStorage.keys
+      if query = params.dig(:search, :query).presence
+        @secret_keys.select! { |s| s.include?(query) }
+      end
+    rescue RuntimeError => e
```

Comment:
so the vault client will raise Vault::HTTPConnectionError.  I didn't really wana catch that in the controller in case the backend is DB.